sycl/include/sycl/stl_wrappers/assert.h

```diff
@@ -10,6 +10,7 @@
 // according to the current state of NDEBUG each time that <cassert> is
 // included.
 
+extern "C++" {
```

Comment:
I don't think that we should change the language linkage of the original `assert` header. It should stay as-is